relay-server/src/actors/spooler/mod.rs

```diff
@@ -490,6 +527,60 @@ impl OnDisk {
         }
     }
 
+    /// Unspools the entire contents of the on-disk spool.
+    async fn delete_and_fetch_all(
+        &mut self,
+        services: &Services,
+    ) -> Result<BTreeMap<QueueKey, Vec<ManagedEnvelope>>, BufferError> {
+        let mut result: BTreeMap<QueueKey, Vec<ManagedEnvelope>> = BTreeMap::new();
+
+        loop {
+            let mut envelopes =
+                sql::delete_and_fetch_all(BATCH_SIZE.min(self.buffer_guard.available() as i64))
```

Comment:
What if `buffer_guard.available()` keeps returning `0`, aren't we just spinning in this loop?

I assume this can't happen because we check for the watermark before calling this function?

Reply:
added another check to make sure we won't have this situation, done in [`b2b35f2`](https://github.com/getsentry/relay/pull/2949/commits/b2b35f2b50f796378f0b84da90c7061eaa9474b6)